### ISSUE TITLE
JsonUtils.arrayPattern is too strict

### ISSUE THREAD

**Issue by jgustie** [lukas-krecan/JsonUnit#47]
I am trying to test some [JSON-LD](http://json-ld.org) output and I am getting caught up on the following JSON:
````
{
  "@id" : "urn:uuid:50aa37c0-eef0-4d72-9f32-17ebbcf17c10",
  "@graph" : [
    { "foo" : "bar" }
  ]
}
````
Using the Fluent assertions, the following fails:
````
assertThatJson(json).node("@graph[0].foo").isEqualTo("bar");
````
Because the string `"@graph[0]"` does not match the array pattern (currently `(\w*)\[(\d+)\]`), in this case the "@" character is not a word character.

Technically, there isn't much that JSON doesn't allow in a name: it is just a string so pretty much any non-control character goes. Edge cases involving the syntax aside (i.e. names that contain square brackets), and borrowing heavily from [here](http://stackoverflow.com/a/3845829/481015) I came up with the following (absolutely terrifying) regex which seems to work:
````
Pattern.compile("((?:(?:[^\"\\\\])|(?:\\\\[\"\\\\bfnrt\\/])|(?:\\\\u\\{XDigit}{4}))*)\\[(\\d+)\\]");
````

**Comment by jgustie** [comment]
It occurred to me while driving home that I considerably overthought this: the code in question doesn't need to valid the JSON, it just needs to extract the name and index. Just replacing the `\w` with `.` is probably a better fix...

**Comment by lukas-krecan** [comment]
Thanks for feedback, fixed in 1.16.1
